pkg/controller/nodegroup/nodegroup_controller.go

```diff
@@ -262,7 +262,7 @@ func (r *ReconcileNodeGroup) createNodeGroupStack(cfnSvc cloudformationiface.Clo
 	templateBody, err := awsHelper.GetCFNTemplateBody(nodeGroupCFNTemplate, nodeGroupTemplateInput{
 		ClusterName:           eks.Spec.ControlPlane.ClusterName,
 		ControlPlaneStackName: eks.GetControlPlaneStackName(),
-		AMI:                   GetAMI(nodegroup.GetVersion(), eks.Spec.Region),
+		AMI:                   GetAMI(*eks.Spec.ControlPlane.Version, eks.Spec.Region),
```

Comment:
I think we can use https://github.com/awslabs/aws-eks-cluster-controller/blob/master/pkg/apis/cluster/v1alpha1/controlplane_types.go#L58 function call to get the eks controlplane version.